ts/features/lollipop/hooks/usePublicKeyState.tsx

```diff
@@ -0,0 +1,32 @@
+import { PublicKey } from "@pagopa/io-react-native-crypto";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import { useEffect, useState } from "react";
+import { useIOSelector } from "../../../store/hooks";
+import { taskGetPublicKey } from "../../../utils/crypto";
+import { lollipopKeyTagSelector } from "../store/reducers/lollipop";
+
+export const usePublicKeyState = () => {
+  const [publicKeyState, setPublicKeyState] = useState<
+    PublicKey | "error" | "retrieving"
+  >("retrieving");
```

Comment:
What do you think about use a enum type instead? Maybe this could be a refactor to do next in io-react-native-crypto package.

Reply:
I would like to propose another approach, using a discriminated union, like:
```
type PKStatus = PKReady | PKChecking | PKError
type PKReady { kind: "ready"; publicKey: PublicKey }
type PKChecking { kind: "checking" }
type PKError { kind: "error"; error: any }

const [publicKeyState, setPublicKeyState] = useState<PKStatus>({ kind: "checking"})
```
What do you think of it?

Reply:
Yep. I suggested anyway to add this in `io-react-native-crypto` package and import from it.